### PR TITLE
Reduce DMFile::restore in DeltaMergeStore::DeltaMergeStore

### PR DESCRIPTION
**Pull request by hongyunyan**
### What problem does this PR solve?

Issue Number: ref #6534

Problem Summary:

### What is changed and how it works?

Avoid do twice DMFile::restore in DeltaMergeStore::DeltaMergeStore to cut down the time during initStores.

### Check List

Tests 

- [ ] Unit test
- [ ] Integration test
- [x] Manual test (add detailed scripts or steps below)
- [ ] No code

I make a cluster and run typical workload to insert data until 600G. Then I restart tiflash and patch the tiflash of this PR, to compare the time spent in DeltaMergeStore::DeltaMergeStore

First Restart : master(14s) vs Ours(9s)
Second Restart: master(9s) vs Ours(6s)

We can find the obvious speed improvement in DeltaMergeStore::DeltaMergeStore.

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note

```release-note
shorten TiFlash startup time
```

### CODE REVIEW

File: dbms/src/Storages/DeltaMerge/StableValueSpace.cpp [hunk 1]
```diff
@@ -107,9 +107,16 @@ StableValueSpacePtr StableValueSpace::restore(DMContext & context, PageIdU64 id)
...
         auto file_id = context.storage_pool.dataReader()->getNormalPageId(page_id);
-        auto file_parent_path = context.path_pool.getStableDiskDelegator().getDTFilePath(file_id);
+        auto path_delegate = context.path_pool.getStableDiskDelegator();
+        auto file_parent_path = path_delegate.getDTFilePath(file_id);
 
         auto dmfile = DMFile::restore(context.db_context.getFileProvider(), file_id, page_id, file_parent_path, DMFile::ReadMetaMode::all());
+        auto res = path_delegate.updateDTFileSize(file_id, dmfile->getBytesOnDisk());
+        if (!res)
+        {
+            // ? 这个值得 throw 吗？还是就打个 ERROR 就可以了呢？
```
Comment (flowbehappy) [inline dbms/src/Storages/DeltaMerge/StableValueSpace.cpp:117]: Use English plz. And exception ok for me.

**Review by flowbehappy — APPROVED** [review]
LGTM with minor comment

File: dbms/src/Storages/DeltaMerge/StableValueSpace.cpp [hunk 2]
```diff
@@ -107,9 +107,15 @@ StableValueSpacePtr StableValueSpace::restore(DMContext & context, PageIdU64 id)
...
-        auto file_parent_path = context.path_pool->getStableDiskDelegator().getDTFilePath(file_id);
+        auto path_delegate = context.path_pool->getStableDiskDelegator();
+        auto file_parent_path = path_delegate.getDTFilePath(file_id);
 
         auto dmfile = DMFile::restore(context.db_context.getFileProvider(), file_id, page_id, file_parent_path, DMFile::ReadMetaMode::all());
+        auto res = path_delegate.updateDTFileSize(file_id, dmfile->getBytesOnDisk());
+        if (!res)
+        {
+            throw Exception(ErrorCodes::LOGICAL_ERROR, "update dt file size failed");
+        }
```
Comment (JaySon-Huang) [inline dbms/src/Storages/DeltaMerge/StableValueSpace.cpp:118]: RUNTIME_CHECK_MSG(res, "update dt file size failed, path={}", ...);

File: dbms/src/Storages/PathPool.cpp [hunk 3]
```diff
@@ -516,6 +516,26 @@ void StableDiskDelegator::addDTFile(UInt64 file_id, size_t file_size, std::strin
     pool.global_capacity->addUsedSize(path, file_size);
 }
 
+bool StableDiskDelegator::updateDTFileSize(UInt64 file_id, size_t file_size)
+{
+    auto iter = pool.dt_file_path_map.find(file_id);
```
Comment (JaySon-Huang) [inline dbms/src/Storages/PathPool.cpp:521]: these codes access to `pool.dt_file_path_map` should be protected by `std::lock_guard lock{pool.mutex};`

**Comment by JaySon-Huang** [comment]
The release note could be "shorten TiFlash startup time"

File: dbms/src/Storages/DeltaMerge/StableValueSpace.cpp [hunk 4]
```diff
@@ -107,9 +107,13 @@ StableValueSpacePtr StableValueSpace::restore(DMContext & context, PageIdU64 id)
...
 
         auto file_id = context.storage_pool->dataReader()->getNormalPageId(page_id);
-        auto file_parent_path = context.path_pool->getStableDiskDelegator().getDTFilePath(file_id);
+        auto path_delegate = context.path_pool->getStableDiskDelegator();
+        auto file_parent_path = path_delegate.getDTFilePath(file_id);
 
         auto dmfile = DMFile::restore(context.db_context.getFileProvider(), file_id, page_id, file_parent_path, DMFile::ReadMetaMode::all());
+        auto res = path_delegate.updateDTFileSize(file_id, dmfile->getBytesOnDisk());
+
+        RUNTIME_CHECK_MSG(res, "update dt file size failed, path={}", file_parent_path);
```
Comment (JaySon-Huang) [inline dbms/src/Storages/DeltaMerge/StableValueSpace.cpp:116]:
```suggestion
        RUNTIME_CHECK_MSG(res, "update dt file size failed, path={}", dmfile->path());
```

**Review by JaySon-Huang — APPROVED** [review]
Rest LGTM

**Pull request merged**